manifests/agent/service.pp

```diff
@@ -29,11 +31,23 @@
 
       $times = ip_to_cron($puppet::runinterval)
 
-      cron { 'puppet':
-        command => $command,
-        user    => root,
-        hour    => $times[0],
-        minute  => $times[1],
+      if $::osfamily == 'windows' {
+        scheduled_task { 'puppet':
+          ensure  => present,
+          enabled => true,
+          command => $command,
+          trigger => {
+            schedule   => daily,
+            start_time => $times[0],
```

Comment:
This is a different schedule. Maybe move the ip_to_cron part to the non-windows part and just try to parse runinterval directly?